rpm/src/main/java/org/eclipse/packager/rpm/info/RpmInformations.java

```diff
@@ -115,66 +149,37 @@ public static RpmInformation makeInformation(final RpmInputStream in) throws IOE
     }
 
     public static List<Dependency> makeDependencies(final InputHeader<RpmTag> header, final RpmTag namesTag, final RpmTag versionsTag, final RpmTag flagsTag) {
-        Object namesVal = header.getTag(namesTag);
-        Object versionsVal = header.getTag(versionsTag);
-        Object flagsVal = header.getTag(flagsTag);
-
-        if (namesVal == null || !(namesVal instanceof String[])) {
-            if (namesVal instanceof String) {
-                namesVal = new String[] { (String) namesVal };
-            } else {
-                return Collections.emptyList();
-            }
-        }
+        final List<String> names = header.getStringList(namesTag);
```

Comment:
Same here, I think we are changing the API. This time it's a public method.

Reply:
I see what you're saying. So the change is that instead of returning an empty list if the types don't match, it will throw a runtime exception.

The types should never fail to match in practice, though.